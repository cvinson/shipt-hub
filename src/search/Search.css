.searchContainer {
  display: flex;
  align-items: center;
  justify-content: center;
}

.search {
  max-width: 300px;
}
