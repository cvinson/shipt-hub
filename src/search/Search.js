import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import TextField from 'material-ui/TextField';
import FlatButton from 'material-ui/FlatButton';
import SearchIcon from 'material-ui/svg-icons/action/search';

export default class Search extends React.Component {
  static props = {
    username: PropTypes.string
  }

  constructor(props) {
    super(props);
    this.state = {
      redirect: false,
      text: ''
    }
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ redirect: false });

    if (this.props.username !== nextProps.username) {
      this.setState({ text: nextProps.username });
    }
  }

  onChange = (event, text) => this.setState({ text });

  onClick = (event) => (this.state.text) && this.setState({ redirect: true });

  onKeyPress = (event) => {
    if (event.key === 'Enter' && this.state.text) {
      this.setState({ redirect: true });
    }
  }

  render() {
    return (
      <div>
        <TextField
          hintText="Search for a Github user"
          id="search-text"
          onChange={this.onChange}
          onKeyPress={this.onKeyPress}
          value={this.state.text} />
        <FlatButton
          backgroundColor="#71b045"
          hoverColor="#000000"
          icon={<SearchIcon />}
          onClick={this.onClick}
          style={{ color: '#ffffff', marginLeft: '5px', minWidth: '40px' }} />
        {this.state.redirect &&
          <Redirect to={{ pathname: `/${this.state.text}`, state: { from: this.props.location } }} />
        }
      </div>
    )
  }
}
