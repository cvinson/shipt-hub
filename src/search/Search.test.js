import React from 'react';
import PropTypes from 'prop-types';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import Search from './Search';

const renderOptions = {
  context: { muiTheme: getMuiTheme() },
  childContextTypes: { muiTheme: PropTypes.object }
};

describe('<Search />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Search username="githubUser" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders a redirect when appropriate', () => {
    const wrapper = shallow(<Search username="githubUser" />, renderOptions);
    wrapper.setState({ redirect: true });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('updates state when onChange is called', () => {
    const wrapper = shallow(<Search username="githubUser" />, renderOptions);
    const search = wrapper.instance();
    search.onChange({}, 'new text');
    expect(wrapper.state(['text'])).toEqual('new text');
  });

  it('updates redirect state when Enter is pressed', () => {
    const wrapper = shallow(<Search username="githubUser" />, renderOptions);
    const search = wrapper.instance();
    wrapper.setState({ text: 'some text' });
    search.onKeyPress({ key: 'Enter' });
    expect(wrapper.state(['redirect'])).toBe(true);
  });

  it('does not update redirect state when a non-Enter key is pressed', () => {
    const wrapper = shallow(<Search username="githubUser" />, renderOptions);
    const search = wrapper.instance();
    wrapper.setState({ text: 'some text' });
    search.onKeyPress({ key: 'a' });
    expect(wrapper.state(['redirect'])).toBe(false);
  });
});
