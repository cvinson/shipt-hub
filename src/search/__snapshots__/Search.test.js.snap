// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders a redirect when appropriate 1`] = `
<div>
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    fullWidth={false}
    hintText="Search for a Github user"
    id="search-text"
    multiLine={false}
    onChange={[Function]}
    onKeyPress={[Function]}
    rows={1}
    type="text"
    underlineShow={true}
    value=""
  />
  <Redirect
    push={false}
    to={
      Object {
        "pathname": "/",
        "state": Object {
          "from": undefined,
        },
      }
    }
  />
</div>
`;

exports[`<Search /> renders correctly 1`] = `
<div>
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    fullWidth={false}
    hintText="Search for a Github user"
    id="search-text"
    multiLine={false}
    onChange={[Function]}
    onKeyPress={[Function]}
    rows={1}
    type="text"
    underlineShow={true}
    value=""
  />
</div>
`;
